src/lib/utils/file-utils.ts

```diff
@@ -1,28 +1,54 @@
 import * as fs from 'fs';
 
+import * as path from 'path';
+
 const UTF_8 = 'utf8';
+const TYPESCRIPT_EXTENSION = 'ts';
+const JAVASCRIPT_EXTENSION = 'js';
 
 export class FileUtils {
-  static isDirectory(path: string): boolean {
-    return fs.existsSync(path) && fs.lstatSync(path).isDirectory();
+  static isDirectory(directoryPath: string): boolean {
+    return fs.existsSync(directoryPath) && fs.lstatSync(directoryPath).isDirectory();
   }
 
-  static isFile(path: string): boolean {
-    return fs.existsSync(path) && fs.lstatSync(path).isFile();
+  static isFile(filePath: string): boolean {
+    return fs.existsSync(filePath) && fs.lstatSync(filePath).isFile();
   }
 
-  static readFile(path: string): string {
-    if (this.isFile(path)) {
-      return fs.readFileSync(path, { encoding: UTF_8 });
+  static readFile(filePath: string): string {
+    if (this.isFile(filePath)) {
+      return fs.readFileSync(filePath, { encoding: UTF_8 });
     }
 
     // TODO: return empty string in this case?
     return '';
   }
 
-  static saveFile(path: string, content: string): void {
-    if (this.isFile(path)) {
-      fs.writeFileSync(path, content, { encoding: UTF_8 });
+  static readDirectory(directoryPath: string, recursive: boolean, filePaths?: string[]): string[] {
+    filePaths = filePaths || [];
+    const files = fs.readdirSync(directoryPath);
+    for (const f in files) {
```

Comment:
Why not for of and then directly use f instead of files[f]?

Reply:
Good catch